File: backend/src/main/scala/microservices/feedback/objects/OrderCancellationRequestPayload.scala
// OrderCancellationRequestPayload：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import com.typesafe.travel.shared.kernel.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
import java.time.Instant

final case class OrderCancellationRequestPayload(
    orderId: String,
    orderTitle: Option[String],
    reason: String,
    requestedRefundAmount: Option[BigDecimal],
    status: OrderCancellationRequestStatus,
    createdAt: String,
    handledAt: Option[String],
    handledBy: Option[String],
    handlerRole: Option[FeedbackSenderRole],
    managerNote: Option[String]
)
object OrderCancellationRequestPayload:
  import ContentSourceJsonCodecs.given
  given sourceEncoder: Encoder[OrderCancellationRequestPayload] = deriveEncoder
  given sourceDecoder: Decoder[OrderCancellationRequestPayload] = deriveDecoder

File: frontend/src/microservices/feedback/objects/OrderCancellationRequestPayload.ts
// 本文件定义 content 模块的 `OrderCancellationRequestPayload`，作为消息载荷并提供 JSON 编解码。

import type { FeedbackSenderRole } from './FeedbackSenderRole'

export type OrderCancellationRequestStatus = 'pending' | 'approved' | 'rejected' | 'needMoreInfo'

export type OrderCancellationRequestPayload = {
  orderId: string
  orderTitle: string | null
  reason: string
  requestedRefundAmount: number | null
  status: OrderCancellationRequestStatus
  createdAt: string
  handledAt: string | null
  handledBy: string | null
  handlerRole: FeedbackSenderRole | null
  managerNote: string | null
}

export const orderCancellationRequestPayloadFromJson = (json: string): OrderCancellationRequestPayload =>
  JSON.parse(json) as OrderCancellationRequestPayload

export const orderCancellationRequestPayloadToJson = (value: OrderCancellationRequestPayload): string =>
  JSON.stringify(value)
